### PR TITLE
[#158285444] Due date CTA

### PR DESCRIPTION
**Pull request by fpersico**
In this PR (a refactor of https://github.com/teamdigitale/italia-app/pull/436) i have also fixed a bit the look and feel of the message list screen.

simulator screen shot - iphone 6 - 2018-09-06 at 14 53 16
simulator screen shot - iphone 6 - 2018-09-06 at 14 53 27
simulator screen shot - iphone 6 - 2018-09-06 at 14 53 44

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#158285444](https://www.pivotaltracker.com/story/show/158285444): Come CIT quando apro il dettaglio di un messaggio che ha una scadenza, voglio poter aggiungere un promemoria nel mio calendario


New dependencies added: <a href='https://www.npmjs.com/package/react-native-add-calendar-event'><code>react-native-add-calendar-event</code></a> and <a href='https://www.npmjs.com/package/react-native-svg'><code>react-native-svg</code></a>.


<h2><a href="https://github.com/vonovak/react-native-add-calendar-event#readme">react-native-add-calendar-event</a></h2>
<p>Author: Vojtech Novak</p>
<p>Description: creating events in react native using the standard iOS / Android dialogs</p>
<p>Homepage: <a href="https://github.com/vonovak/react-native-add-calendar-event#readme">https://github.com/vonovak/react-native-add-calendar-event#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 1 year ago</td></tr><tr><td>Last Updated</td><td>8 days ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>18</td></tr><tr><td>Keywords</td><td>react-native</td></tr>
</table>
This README is too long to show.



<h2><a href="https://github.com/magicismight/react-native-svg#readme">react-native-svg</a></h2>
<p>Author: Unknown</p>
<p>Description: SVG library for react-native</p>
<p>Homepage: <a href="https://github.com/magicismight/react-native-svg#readme">https://github.com/magicismight/react-native-svg#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 3 years ago</td></tr><tr><td>Last Updated</td><td>about 1 month ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>4</td></tr><tr><td>Releases</td><td>104</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/color'><code>color</code></a>, <a href='https://www.npmjs.com/package/lodash'><code>lodash</code></a> and <a href='https://www.npmjs.com/package/pegjs'><code>pegjs</code></a></td></tr><tr><td>Keywords</td><td>react-component, react-native, ios, android, SVG, ART, VML and gradient</td></tr>
</table>
This README is too long to show.

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by cloudify** [comment]
@fpersico gorgeous! can you also add a screen with a message with both a reminder and payment info?

File: ts/components/messages/MessageComponent.tsx [hunk 1]
```diff
@@ -47,6 +49,44 @@ export type Props = OwnProps &
   ReduxInjectedProps &
   NavigationInjectedProps;
 
+const TouchableFeedbackComponent = Platform.select({
```
Comment (cloudify) [inline ts/components/messages/MessageComponent.tsx]: is this something we should do for other interactions too?

Comment (cloudify) [reply ts/components/messages/MessageComponent.tsx]: can you add a little comment explaining why this is needed?

File: ts/components/messages/MessageComponent.tsx [hunk 2]
```diff
@@ -57,12 +97,26 @@ class MessageComponent extends React.Component<Props> {
...
+      due_date,
       payment_data,
       sender_service_id
     } = message;
 
     const senderService = this.props.serviceByIdMap[sender_service_id];
 
+    // Create an action that open the Calendar to let the user add an event
+    const dispatchReminderAction = fromNullable(due_date)
+      .map(_ => () => {
```
Comment (cloudify) [inline ts/components/messages/MessageComponent.tsx]: the `return` is not needed

```ts
() => AddCalendarEvent.presentEventCreatingDialog(...);
```

**Pull request merged**